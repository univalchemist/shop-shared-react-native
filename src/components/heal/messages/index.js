import en_HK from './en-HK.json';
import zh_HK from './zh-HK.json';

const messages = {
  'en-HK': en_HK,
  'zh-HK': zh_HK,
};

export default messages;
export { en_HK, zh_HK };
