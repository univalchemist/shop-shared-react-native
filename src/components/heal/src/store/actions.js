import * as types from './types';
import moment from 'moment';

export const getDoctors = specialityCode => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId } = getState().user;
    const { doctorData } = getState().heal;
    let page = doctorData.page + 1;
    if (doctorData.specialityCode !== specialityCode) {
      page = 1;
    }
    const { data } = await api.getDoctors({ clientId, page, specialityCode });
    return { ...data, specialityCode };
  };
  return dispatch({
    type: types.GET_DOCTORS,
    payload: getPromise(),
  });
};

export const getClinics = ({
  searchTerm,
  searchBy,
  forceReload,
} = {}) => async (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId } = getState().user;
    const { location, clinicData } = getState().heal;
    let page = clinicData.page + 1;
    if (
      searchTerm !== clinicData.searchTerm ||
      searchBy !== clinicData.searchBy
    )
      page = 1;

    if (forceReload) page = 1;

    const response = await api.getClinicsWithObject({
      clientId,
      lat: location.latitude,
      lng: location.longitude,
      page,
      searchTerm,
      searchBy,
    });
    response.data.items = transformClinicForCluster(response.data.items);
    return { ...response.data, searchTerm, searchBy, forceReload };
  };

  return dispatch({
    type: types.FETCH_CLINICS,
    payload: getPromise(),
  });
};

export const fetchLocation = () => async (dispatch, getState, { api }) => {
  const position = {
    coords: {
      latitude: 0,
      longitude: 0,
    },
  };
  const getPromise = async () => {
    return new Promise(resolve => {
      navigator.geolocation.getCurrentPosition(
        position => resolve(position),
        error => resolve(position),
      );
    });
  };

  return dispatch({
    type: types.FETCH_LOCATION,
    payload: getPromise(),
  });
};

export const getSpecialities = () => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId } = getState().user;
    const { data } = await api.getSpecialities(clientId);
    return data;
  };
  return dispatch({
    type: types.GET_SPECIALITIES,
    payload: getPromise(),
  });
};

export const getDoctorsInfo = clinic => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId, userId } = getState().user;
    const clinicProviderId = clinic.clinicProviderId;
    const clinicId = clinic.id;
    let promiseArray = [];

    for (let i = 0; i < clinic.doctors.length; i += 1) {
      const doctor = clinic.doctors[i];
      promiseArray.push(
        api.getDoctorInfo({
          clientId,
          userId,
          clinicProviderId,
          clinicId,
          doctorId: doctor.id,
        }),
      );
    }
    const data = await Promise.all(promiseArray);
    return data;
  };
  return dispatch({
    type: types.GET_DOCTORS_INFO,
    payload: getPromise(),
  });
};

export const setDetailsClinic = clinic => (dispatch, getState, { api }) => {
  return dispatch({
    type: types.SET_DETAILS_CLINIC,
    payload: clinic,
  });
};

export const search = ({ searchTerm, latitude, longitude }) => (
  dispatch,
  getState,
  { api },
) => {
  const searchClinics = async ({
    searchBy,
    searchTerm,
    searchType,
    lat = null,
    lng = null,
    nearBy = null,
  }) => {
    const { clientId } = getState().user;
    const { data } = await api.getClinicsWithObject({
      clientId,
      lat,
      lng,
      page: 1,
      itemPerPage: types.GET_ALL,
      nearBy,
      searchBy,
      searchTerm,
    });
    return { ...data, searchType };
  };

  const searchDoctors = async ({
    searchBy,
    searchTerm,
    specialityCode,
    searchType,
  }) => {
    const { clientId } = getState().user;
    const { data } = await api.getDoctors({
      clientId,
      page: 1,
      itemPerPage: types.GET_ALL,
      searchBy,
      searchTerm,
      specialityCode,
    });
    if (specialityCode) return { ...data, specialityCode, searchType };
    else return { ...data, searchType };
  };

  const doSearch = async () => {
    const { specialities, specialitiesByCode } = getState().heal;

    let data = [];
    let raw;
    let specialitySearchResult = [];

    if (searchTerm === types.SearchItemType.NearMe) {
      raw = await searchClinics({
        searchType: types.SearchItemType.NearMe,
        nearBy: types.UNIFY_SEARCH_DISTANCE,
        lat: latitude,
        lng: longitude,
      });
    } else {
      const filteredSpecialities = specialities.filter(s =>
        s.name.toLowerCase().includes(searchTerm.toLowerCase()),
      );

      raw = await Promise.all([
        searchDoctors({
          searchTerm,
          searchBy: types.DoctorSearchType.Name,
          searchType: types.SearchItemType.Doctor,
        }),
        ...filteredSpecialities.map(f =>
          searchDoctors({
            specialityCode: f.code,
            searchType: types.SearchItemType.Speciality,
          }),
        ),
        searchClinics({
          searchBy: types.ClinicSearchType.Name,
          searchTerm,
          searchType: types.SearchItemType.Clinic,
        }),
        searchClinics({
          searchBy: types.ClinicSearchType.Location,
          searchTerm,
          searchType: types.SearchItemType.Location,
        }),
      ]);
    }

    for (let i = 0; i < raw.length; i++) {
      const res = raw[i];
      switch (res.searchType) {
        case types.SearchItemType.Doctor:
          if (res.items?.length > 0)
            data.push({
              title: 'heal.UnifySearch.DoctorName',
              data: res.items.map(d => ({
                title: d.name,
                subtitle: null,
                data: { ...d, type: types.SearchItemType.Doctor },
              })),
            });
          break;
        case types.SearchItemType.Speciality:
          if (res.totalCount > 0) {
            specialitySearchResult.push({
              title: specialitiesByCode[res.specialityCode].name,
              subtitle: res.totalCount,
              data: {
                specialityCode: res.specialityCode,
                type: types.SearchItemType.Speciality,
              },
            });
          }
          const s = data.find(d => d.title === 'heal.UnifySearch.Speciality');
          if (!s && specialitySearchResult.length > 0)
            data.push({
              title: 'heal.UnifySearch.Speciality',
              data: specialitySearchResult,
            });
          break;
        case types.SearchItemType.Location:
          if (res.totalCount > 0) {
            let location = {};
            const items = res.items;
            for (let i = 0; i < items.length; i++) {
              if (!location[items[i].district]) location[items[i].district] = 0;
              location[items[i].district] += 1;
            }
            data.push({
              title: 'heal.UnifySearch.Location',
              data: Object.keys(location).map(k => ({
                title: k,
                subtitle: location[k],
                data: {
                  districtCode: k,
                  type: types.SearchItemType.Location,
                },
              })),
            });
          }
          break;
        case types.SearchItemType.Clinic:
          if (res.items?.length > 0)
            data.push({
              title: 'heal.UnifySearch.ClinicName',
              data: res.items.map(c => ({
                title: c.name,
                subtitle: null,
                data: { ...c, type: types.SearchItemType.Clinic },
              })),
            });
          break;
      }
    }

    console.log(data);
    return data;
  };

  return dispatch({
    type: types.SEARCH,
    payload: doSearch(),
  });
};

export const clearSearch = () => (dispatch, getState, { api }) => {
  return dispatch({ type: types.CLEAR_SEARCH });
};

const transformClinicForCluster = clinics => {
  const data = clinics.map(clinic => ({
    ...clinic,
    location: { longitude: clinic.longitude, latitude: clinic.latitude },
  }));
  return data;
};

export const updateSelectedClinic = clinic => dispatch => {
  return dispatch({
    type: types.UPDATE_SELECTED_CLINIC,
    payload: clinic,
  });
};

export const updateSelectedClinics = clinics => dispatch => {
  return dispatch({
    type: types.UPDATE_SELECTED_CLINICS,
    payload: clinics,
  });
};

export const updateClinicPage = () => dispatch => {
  return dispatch({
    type: types.UPDATE_CLINIC_PAGE,
  });
};

export const scanQRCode = clinicQrCode => (dispatch, getState, { api }) => {
  const scan = async () => {
    const { clientId, userId } = getState().user;
    const { data } = await api.scanQRCode({ clientId, userId, clinicQrCode });
    return { ...data, clinicQrCode };
  };
  return dispatch({
    type: types.SCAN_QR_CODE,
    payload: scan(),
  });
};

export const checkInWalkIn = ({ clinicProviderId, clinicId, doctorId }) => (
  dispatch,
  getState,
  { api },
) => {
  const getPromise = async () => {
    const { clientId, userId } = getState().user;
    const { data } = api.checkInWalkIn({
      clientId,
      userId,
      clinicProviderId,
      clinicId,
      doctorId,
    });
    return data;
  };

  return dispatch({
    type: types.CHECK_IN_WALK_IN,
    payload: getPromise(),
  });
};

export const getWalkInSelectDoctors = clinicQrCode => (
  dispatch,
  getState,
  { api },
) => {
  const getPromise = async () => {
    const { clientId, userId } = getState().user;
    const { data } = await api.getWalkInSelectDoctors({
      clientId,
      userId,
      clinicQrCode,
    });
    return data;
    //     return {
    //       clinic: {
    //         id: 373,
    //         name: 'CXA (SG) Testing Clinic 1',
    //         latitude: 1.303007,
    //         longitude: 103.796334,
    //         qrCode: '304',
    //         district: 'CAUSEWAY BAY',
    //         area: null,
    //       },
    //       doctors: [
    //         {
    //           isOpen: false,
    //           isRemoteFull: false,
    //           isRemoteTicketEnabled: true,
    //           isAppointmentEnabled: true,
    //           position: 0,
    //           estimatedConsultationTime: '1:48 pm',
    //           ticketDepositAmount: 0.0,
    //           canGetRefundOnCancellation: false,
    //           cancellationGraceMinutes: 0,
    //           id: 355,
    //           name: 'Chan Tai Man',
    //           specialityCode: 'generalpractitio',
    //           gender: 'Male',
    //           email: null,
    //           imageUrl: null,
    //           introduction: null,
    //         },
    //         {
    //           isOpen: false,
    //           isRemoteFull: false,
    //           isRemoteTicketEnabled: true,
    //           isAppointmentEnabled: true,
    //           position: 0,
    //           estimatedConsultationTime: '1:48 pm',
    //           ticketDepositAmount: 0.0,
    //           canGetRefundOnCancellation: false,
    //           cancellationGraceMinutes: 0,
    //           id: 354,
    //           name: 'CXA2',
    //           specialityCode: 'generalpractitio',
    //           gender: 'Male',
    //           email: null,
    //           imageUrl: null,
    //           introduction: null,
    //         },
    //       ],
    //     };
  };

  return dispatch({
    type: types.GET_WALKIN_SELECTS,
    payload: getPromise(),
  });
};

export const getMedicalProfile = () => (dispatch, getState, { api }) => {
  const fetchProfile = async () => {
    const { clientId } = getState().user;
    const {
      values: { memberId },
    } = getState().form.checkinForm;
    const { data } = await api.getMedicalProfile(clientId, memberId);
    return data;
  };
  return dispatch({
    type: types.GET_MEDICAL_PROFILE,
    payload: fetchProfile(),
  });
};

export const updateMedicalProfile = () => (dispatch, getState, { api }) => {
  const updateProfile = async () => {
    const { clientId } = getState().user;
    const { values } = getState().form.checkinForm;
    const { data } = await api.updateMedicalProfile({
      clientId,
      userId: values.memberId,
      ...values,
    });
    return data;
  };
  return dispatch({
    type: types.UPDATE_MEDICAL_PROFILE,
    payload: updateProfile(),
  });
};

export const getAppointmentList = () => (dispatch, getState, { api }) => {
  const fetchAppointment = async () => {
    const { clientId, userId } = getState().user;
    const { data } = await api.getAppointmentList(clientId, userId);
    dispatch({
      type: types.GET_APPOINTMENT_LIST,
      payload: data,
    });
  };
  fetchAppointment();
};

export const acceptAppointment = (confirm, appointmentId) => (
  dispatch,
  getState,
  { api },
) => {
  const updateAppointment = async () => {
    const { clientId, userId } = getState().user;
    const { data } = await api.acceptAppointment({
      clientId,
      userId: userId,
      appointmentId,
      isConfirmed: confirm,
    });
    return data;
  };
  return dispatch({
    type: types.ACCEPT_APPOINTMENT,
    payload: updateAppointment(),
  });
};

export const getRemoteTickets = () => (dispatch, getState, { api }) => {
  const getTickets = async () => {
    const { clientId, userId } = getState().user;
    const { data } = await api.getRemoteTickets(clientId, userId);
    return data;
  };
  return dispatch({
    type: types.GET_REMOTE_TICKETS,
    payload: getTickets(),
  });
};

export const getMembers = () => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId, userId } = getState().user;
    const response = await api.fetchMemberProfile(clientId, userId);
    return response.data;
  };
  return dispatch({
    type: types.GET_MEMBER,
    payload: getPromise(),
  });
};

export const requestAppointment = (
  clinicProviderId,
  doctorId,
  clinicId,
  dates,
) => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId, userId } = getState().user;
    let datesObj = {};
    dates.map((item, index) => {
      if (index === 0) {
        datesObj.firstPreferredTime = moment(item.date).format();
        datesObj.firstSession = item.time;
      }
      if (index === 1) {
        datesObj.secondPreferredTime = moment(item.date).format();
        datesObj.secondSession = item.time;
      }
      if (index === 2) {
        datesObj.thirdPreferredTime = moment(item.date).format();
        datesObj.thirdSession = item.time;
      }
    });
    const appointmentData = {
      clinicProviderId: clinicProviderId,
      doctorId: doctorId,
      clinicId: clinicId,
      ...datesObj,
    };
    const { data } = await api.requestAppointment({
      clientId,
      userId,
      appointmentData,
    });
    return data;
  };
  return dispatch({
    type: types.REQUEST_APPOINTMENT,
    payload: getPromise(),
  });
};
