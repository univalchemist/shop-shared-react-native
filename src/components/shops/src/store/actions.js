/* istanbul ignore file */

export * from './config/actions';
export * from './filters/actions';
export * from './home/actions';
export * from './review/actions';
export * from './cart/actions';
export * from './orderhistory/actions';
export * from './deliveryAddress/actions';
