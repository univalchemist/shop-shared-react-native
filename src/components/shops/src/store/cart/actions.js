import {
  ADD_TO_CART,
  GET_CART,
  GET_CART_BILLING,
  GET_CART_SHIPPING,
  GET_CART_TOTALS,
  REMOVE_ITEM_FROM_CART,
} from './types';

export const addToCart = ({ sku, quantity }) => (
  dispatch,
  getState,
  { api },
) => {
  const getPromise = async () => {
    const { userId, clientId } = getState().user;
    const { data } = await api.addToCart({
      clientId,
      userId,
      sku,
      quantity,
    });
    return data;
  };

  return dispatch({
    type: ADD_TO_CART,
    payload: getPromise(),
  });
};

export const getCart = () => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { userId, clientId } = getState().user;
    const { data } = await api.getCart(clientId, userId);
    return data;
  };

  return dispatch({
    type: GET_CART,
    payload: getPromise(),
  });
};

export const getCartTotals = () => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { userId, clientId } = getState().user;
    const { data } = await api.getCartTotals(clientId, userId);
    return data;
  };

  return dispatch({
    type: GET_CART_TOTALS,
    payload: getPromise(),
  });
};

export const getCartBilling = () => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { userId, clientId } = getState().user;
    const { data } = await api.getCartBilling(clientId, userId);
    return data;
  };

  return dispatch({
    type: GET_CART_BILLING,
    payload: getPromise(),
  });
};

export const getCartShipping = () => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { userId, clientId } = getState().user;
    const { data } = await api.getCartShipping(clientId, userId);
    return data;
  };

  return dispatch({
    type: GET_CART_SHIPPING,
    payload: getPromise(),
  });
};

export const removeItemFromCart = sku => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { userId, clientId } = getState().user;
    const { data } = await api.removeItemFromCart(clientId, userId, sku);
    return data;
  };

  return dispatch({
    type: REMOVE_ITEM_FROM_CART,
    payload: getPromise(),
  });
};
