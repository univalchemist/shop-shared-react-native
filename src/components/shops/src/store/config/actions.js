import {
  GET_STORE_CONFIGS,
  GET_CATEGORIES,
  GET_SORTINGS,
  GET_COUNTRIES,
} from './types';

export const getStoreConfigs = () => async (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId } = getState().user;

    const { data } = await api.getStoreConfigs(clientId);
    return data && data[0];
  };

  return dispatch({
    type: GET_STORE_CONFIGS,
    payload: getPromise(),
  });
};

export const getCategories = () => async (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId } = getState().user;

    const { data } = await api.getCategories(clientId);
    return data && data[0];
  };

  return dispatch({
    type: GET_CATEGORIES,
    payload: getPromise(),
  });
};

export const getSortings = () => async (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId } = getState().user;

    const { data } = await api.getSortings(clientId);
    return data;
  };

  return dispatch({
    type: GET_SORTINGS,
    payload: getPromise(),
  });
};

export const getCountries = () => async (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId } = getState().user;

    const { data } = await api.getCountries(clientId);
    return data;
  };

  return dispatch({
    type: GET_COUNTRIES,
    payload: getPromise(),
  });
};
