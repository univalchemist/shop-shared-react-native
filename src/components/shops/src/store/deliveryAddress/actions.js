import {
  GET_DELIVERY_ADDRESSES,
  DELETE_DELIVERY_ADDRESS,
  ADD_DELIVERY_ADDRESS,
  UPDATE_DELIVERY_ADDRESS,
} from './types';

export const getDeliveryAddresses = () => (dispatch, getState, { api }) => {
  const getPromise = async () => {
    const { clientId, userId } = getState().user;
    const { data } = await api.getDeliveryAddresses(clientId, userId);

    return data;
  };

  return dispatch({
    type: GET_DELIVERY_ADDRESSES,
    payload: getPromise(),
  });
};

export const deleteDeliveryAddress = addressId => (
  dispatch,
  getState,
  { api },
) => {
  const getPromise = async () => {
    const { clientId, userId } = getState().user;
    const { data } = await api.deleteDeliveryAddress(
      clientId,
      userId,
      addressId,
    );

    return data;
  };

  return dispatch({
    type: DELETE_DELIVERY_ADDRESS,
    payload: getPromise(),
    meta: { addressId },
  });
};

export const addDeliveryAddress = addressData => (
  dispatch,
  getState,
  { api },
) => {
  const getPromise = async () => {
    const { clientId, userId, membersMap } = getState().user;
    const user = membersMap[userId];
    const address = {
      ...addressData,
      telephone: user.contactNumber || '24124234', // TODO: why do we don't have these fields in address form
      firstName: user.firstName,
      lastName: user.lastName,
    };

    const { data } = await api.addDeliveryAddress(clientId, userId, address);

    return data;
  };

  return dispatch({
    type: ADD_DELIVERY_ADDRESS,
    payload: getPromise(),
  });
};

export const updateDeliveryAddress = addressData => (
  dispatch,
  getState,
  { api },
) => {
  const getPromise = async () => {
    const { clientId, userId } = getState().user;
    const { data } = await api.updateDeliveryAddress(
      clientId,
      userId,
      addressData.id,
      addressData,
    );

    return data;
  };

  return dispatch({
    type: UPDATE_DELIVERY_ADDRESS,
    payload: getPromise(),
  });
};
