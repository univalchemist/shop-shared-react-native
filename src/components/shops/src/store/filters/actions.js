import {
  UPDATE_SELECTED_CATEGORIES,
  UPDATE_SORT_TYPE,
  UPDATE_FILTER_TYPES,
  UPDATE_SEARCH_STRING,
} from './types';

export const updateSelectedCategories = categories => ({
  type: UPDATE_SELECTED_CATEGORIES,
  payload: categories,
});

export const updateSortType = sortType => ({
  type: UPDATE_SORT_TYPE,
  payload: sortType,
});

export const updateFilterTypes = filterTypes => ({
  type: UPDATE_FILTER_TYPES,
  payload: filterTypes,
});

export const updateSearchString = searchString => ({
  type: UPDATE_SEARCH_STRING,
  payload: searchString,
});
