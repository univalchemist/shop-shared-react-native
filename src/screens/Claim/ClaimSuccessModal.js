import React, { useCallback } from 'react';
import { Box, Button, Image, StatusPanel } from '@wrappers/components';
import { CLAIM_PATIENT_DETAILS, CLAIMS_LIST } from '@routes';
import { useIntl, useBackButtonHandler } from '@wrappers/core/hooks';
import { StackActions } from '@react-navigation/native';
import { claimSuccessImage } from '@images/claim';
import { getClaims, updateClaimFilters } from '@store/claim/actions';
import { connect } from 'react-redux';
import { compose } from 'redux';

const ClaimSuccessModal = ({ updateClaimFilters, getClaims, navigation }) => {
  const intl = useIntl();

  const handleViewSubmittedClaims = async () => {
    await updateClaimFilters([]);
    getClaims();
    navigation.dispatch(StackActions.popToTop());
  };

  const handleMakeAnotherClaimPress = async () => {
    await navigation.reset({
      routes: [{ name: CLAIMS_LIST }],
    });

    navigation.navigate(CLAIM_PATIENT_DETAILS);
  };

  const onBackButtonPressAndroid = useCallback(() => {
    navigation.dispatch(StackActions.popToTop());
    return true;
  });

  useBackButtonHandler(onBackButtonPressAndroid);

  return (
    <StatusPanel
      image={
        <Image
          source={claimSuccessImage}
          maxWidth={215}
          maxHeight={215}
          resizeMode="contain"
        />
      }
      heading={intl.formatMessage({ id: 'claim.claimSuccessModalTitle' })}
      description={intl.formatMessage({ id: 'claim.claimSuccessModalText' })}
      disclaimer={intl.formatMessage({
        id: 'claim.claimSuccessDisclaimerText',
      })}
      actions={
        <>
          <Box>
            <Button
              primary
              title={intl.formatMessage({
                id: 'claim.viewSubmittedClaimsText',
              })}
              onPress={handleViewSubmittedClaims}
            />
          </Box>
          <Box mt={3}>
            <Button
              secondary
              title={intl.formatMessage({ id: 'claim.makeAnotherClaimText' })}
              onPress={handleMakeAnotherClaimPress}
            />
          </Box>
        </>
      }
    />
  );
};

const enhance = compose(connect(null, { getClaims, updateClaimFilters }));

export default enhance(ClaimSuccessModal);
