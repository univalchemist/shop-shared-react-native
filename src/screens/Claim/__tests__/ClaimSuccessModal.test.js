import React from 'react';
import { renderForTest } from '@testUtils';
import {
  fireEvent,
  flushMicrotasksQueue,
  act,
} from 'react-native-testing-library';
import MockBackHandler from 'react-native/Libraries/Utilities/__mocks__/BackHandler';
import { BackHandler } from 'react-native';
import messages from '@messages/en-HK.json';
import navigation from '@testUtils/__mocks__/navigation';
import ClaimSuccessModal from '../ClaimSuccessModal';
import { getClaims, updateClaimFilters } from '@store/claim/actions';
import { CLAIMS_LIST } from '@routes';

jest.mock('@store/claim/actions', () => ({
  getClaims: jest.fn(() => {
    return {
      type: 'TEST_CALL',
      payload: {},
    };
  }),
  updateClaimFilters: jest.fn(() => {
    return {
      type: 'TEST_CALL',
      payload: {},
    };
  }),
}));

jest.mock(
  'react-native/Libraries/Utilities/BackHandler',
  () => MockBackHandler,
);

describe('ClaimSuccessModal', () => {
  const renderClaimSuccessModal = () =>
    renderForTest(
      <ClaimSuccessModal
        navigation={navigation}
        updateClaimFilters={updateClaimFilters}
        getClaims={getClaims}
      />,
    );
  beforeEach(() => {
    jest.clearAllMocks();
    navigation.clearAll();
  });

  it.skip('should render correctly', () => {
    const component = renderClaimSuccessModal();
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('upon pressing view submitted claims', () => {
    it('should clear existing claim filters if any', async () => {
      const component = renderClaimSuccessModal();
      const button = component.getByProps({
        title: messages['claim.viewSubmittedClaimsText'],
      });

      act(() => {
        fireEvent.press(button);
      });
      await flushMicrotasksQueue();

      expect(updateClaimFilters).toHaveBeenCalledWith([]);
    });

    it('should get claims', async () => {
      const component = renderClaimSuccessModal();
      const button = component.getByProps({
        title: messages['claim.viewSubmittedClaimsText'],
      });

      act(() => {
        fireEvent.press(button);
      });
      await flushMicrotasksQueue();

      expect(getClaims).toHaveBeenCalledTimes(1);
    });

    it('should navigate to CLAIMS_LIST', async () => {
      const component = renderClaimSuccessModal();
      const button = component.getByProps({
        title: messages['claim.viewSubmittedClaimsText'],
      });
      act(() => {
        fireEvent.press(button);
      });
      await flushMicrotasksQueue();

      expect(navigation.dispatch).toHaveBeenCalled();
    });
  });

  it('should navigate to CLAIMS and then CLAIM_PATIENT_DETAILS upon pressing make another claim', async () => {
    const component = renderClaimSuccessModal();
    const button = component.getByProps({
      title: messages['claim.makeAnotherClaimText'],
    });
    act(() => {
      fireEvent.press(button);
    });
    await flushMicrotasksQueue();

    expect(navigation.reset).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledTimes(1);
  });

  it('should navigate to CLAIMS on BackButtonPress', async () => {
    const component = renderClaimSuccessModal();

    await flushMicrotasksQueue();
    BackHandler.mockPressBack();

    component.unmount();

    expect(navigation.dispatch).toHaveBeenCalled();
  });
});
