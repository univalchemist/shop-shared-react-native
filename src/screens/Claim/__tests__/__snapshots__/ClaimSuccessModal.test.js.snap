// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimSuccessModal should render correctly 1`] = `
<RCTScrollView
  centerContent={true}
>
  <View>
    <View
      alignItems="center"
      display="flex"
      p={4}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "paddingBottom": 32,
            "paddingLeft": 32,
            "paddingRight": 32,
            "paddingTop": 32,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Image
          maxHeight={215}
          maxWidth={215}
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/images/claimSuccess.png",
            }
          }
          style={
            Array [
              Object {
                "maxHeight": 215,
                "maxWidth": 215,
              },
            ]
          }
        />
      </View>
      <View
        mt={4}
        style={
          Array [
            Object {
              "marginTop": 32,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            color="gray.0"
            fontFamily="default"
            fontSize={32}
            fontWeight="light"
            letterSpacing={-1.5}
            lineHeight={37}
            style={
              Array [
                Object {
                  "color": "#212121",
                  "fontFamily": "Open Sans",
                  "fontSize": 32,
                  "fontWeight": "300",
                  "letterSpacing": -1.5,
                  "lineHeight": 37,
                  "textAlign": "center",
                },
              ]
            }
            textAlign="center"
          >
            Claim submitted
          </Text>
        </View>
        <View
          mt={3}
          style={
            Array [
              Object {
                "marginTop": 16,
              },
            ]
          }
        >
          <Text
            color="gray.1"
            fontFamily="default"
            fontSize={16}
            lineHeight={22}
            style={
              Array [
                Object {
                  "color": "#646464",
                  "fontFamily": "Open Sans",
                  "fontSize": 16,
                  "lineHeight": 22,
                  "textAlign": "center",
                },
              ]
            }
            textAlign="center"
          >
            Thank you. Please keep your original documents for at least 6 months.
          </Text>
        </View>
      </View>
      <View
        mt={4}
        style={
          Array [
            Object {
              "marginTop": 32,
              "width": "100%",
            },
          ]
        }
        width="100%"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "borderRadius": 3,
              }
            }
          >
            <View
              accessibilityLabel="View Submitted Claims"
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#50ABAD",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 48,
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Open Sans",
                      "fontSize": 16,
                      "lineHeight": 20,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  View Submitted Claims
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          mt={3}
          style={
            Array [
              Object {
                "marginTop": 16,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderRadius": 3,
              }
            }
          >
            <View
              accessibilityLabel="Make another claim"
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#f9f9f9",
                    "borderColor": "#646464",
                    "borderRadius": 3,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "height": 48,
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#212121",
                      "fontFamily": "Open Sans",
                      "fontSize": 16,
                      "lineHeight": 20,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  Make another claim
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
